src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -3024,7 +3022,33 @@ public ConstructorEdit(INamedTypeSymbol oldType)
                             Contract.ThrowIfNull(oldSymbol);
 
                             AnalyzeSymbolUpdate(oldSymbol, newSymbol, edit.NewNode, newCompilation, editScript.Match, capabilities, diagnostics, semanticEdits, syntaxMap, cancellationToken);
-                            if (newSymbol is INamedTypeSymbol or IFieldSymbol or IPropertySymbol or IEventSymbol or IParameterSymbol or ITypeParameterSymbol)
+
+                            if (newSymbol is INamedTypeSymbol or IFieldSymbol or IParameterSymbol or ITypeParameterSymbol)
+                            {
+                                continue;
+                            }
+
+                            // For renames where the symbol allows deletion, we don't create an update edit, we create a delete
+                            // and an add. During emit an empty body will be created for the old name. Sometimes
+                            // when members are moved between documents in partial classes, they can appear as renames, so
+                            // we also check that the old symbol can't be resolved in the new compilation
+                            if (oldSymbol.Name != newSymbol.Name &&
+                                AllowsDeletion(oldSymbol) &&
+                                SymbolKey.Create(oldSymbol, cancellationToken) is { } oldSymbolKey &&
```

Comment:
>is { } oldSymbolKey [](http://example.com/codeflow?start=79&length=19)

SymbolKey.Create never returns null. This would do:
```
SymbolKey.Create(oldSymbol, cancellationToken).Resolve(...)
```